File: qosten-react/extract_cq.py
import json

input_file = r"C:\Users\DFIT\Downloads\question-bank-full-2026-06-05.json"
output_file = "higher_math_cq.txt"
target_subject = "?????? ????"
target_type = "cq"

try:
    with open(input_file, "r", encoding="utf-8") as f:
        data = json.load(f)
    
    filtered_data = [item for item in data if item.get("subject") == target_subject and item.get("type") == target_type]
    
    with open(output_file, "w", encoding="utf-8") as f:
        for item in filtered_data:
            f.write(json.dumps(item, ensure_ascii=False, indent=2) + "\n\n" + "-"*40 + "\n\n")
    
    print(f"Successfully extracted {len(filtered_data)} questions to {output_file}")

except Exception as e:
    print(f"Error: {e}")

File: qosten-react/extract_cq.js
const fs = require("fs");

const inputFile = "C:\\Users\\DFIT\\Downloads\\question-bank-full-2026-06-05.json";
const outputFile = "higher_math_cq.txt";
const targetSubject = "?????? ????";
const targetType = "cq";

try {
    console.log("Reading file...");
    const data = fs.readFileSync(inputFile, "utf8");
    const json = JSON.parse(data);
    
    console.log("Filtering data...");
    const filtered = json.filter(item => item.subject === targetSubject && item.type === targetType);
    
    console.log(`Found ${filtered.length} matches. Writing to file...`);
    const outputContent = filtered.map(item => JSON.stringify(item, null, 2) + "\n\n" + "-".repeat(40) + "\n\n").join("");
    fs.writeFileSync(outputFile, outputContent, "utf8");
    
    console.log(`Successfully extracted ${filtered.length} questions to ${outputFile}`);
} catch (err) {
    console.error("Error:", err.message);
    process.exit(1);
}
